app/src/androidTest/java/com/github/se/cyrcle/ui/profile/ViewProfileScreenTest.kt

```diff
@@ -13,203 +25,289 @@ import org.mockito.Mockito.mock
-      composeTestRule.onAllNodesWithTag("FavoriteParkingsTitle").fetchSemanticsNodes().isNotEmpty()
-    }
+  fun testStarIconClickDisplaysConfirmationDialog() {
+    composeTestRule.waitForIdle()
 
-    // Check favorite parkings section exists
-    composeTestRule.onNodeWithTag("FavoriteParkingsTitle").assertExists()
+    composeTestRule.onNodeWithTag("FavoriteToggle_0").performClick()
+    composeTestRule.waitForIdle()
 
-    // Verify the favorite parking items
-    composeTestRule.onNodeWithTag("FavoriteParkingList").assertExists()
+    composeTestRule.onNodeWithText("Remove favorite").assertIsDisplayed()
     composeTestRule
-        .onAllNodesWithTag("ParkingItem_")
-        .assertAll(hasTextExactly("Parking Central", "Parking Station", "Parking Mall"))
-    composeTestRule.onAllNodesWithTag("FavoriteToggle_").assertAll(isToggleable())
+        .onNodeWithText("Are you sure you want to remove Rue de la paix from your favorites?")
+        .assertIsDisplayed()
   }
 
   @Test
-  fun testProfileImageInteraction() {
-    composeTestRule.waitUntil(timeoutMillis = 5000) {
-      composeTestRule.onAllNodesWithTag("ProfileImage").fetchSemanticsNodes().isNotEmpty()
-    }
+  fun testConfirmRemoveFavoriteParking() {
+    composeTestRule.waitForIdle()
 
-    // Verify profile image exists
-    composeTestRule.onNodeWithTag("ProfileImage").assertExists()
+    // Click the star icon to show the confirmation dialog
+    composeTestRule.onNodeWithTag("FavoriteToggle_0").performClick()
+    composeTestRule.waitForIdle()
 
-    // Enter edit mode
-    composeTestRule.onNodeWithTag("EditButton").performClick()
+    // Confirm removal
+    composeTestRule.onNodeWithText("Remove").performClick()
+    composeTestRule.waitForIdle()
 
-    composeTestRule.waitUntil(timeoutMillis = 5000) {
-      composeTestRule.onAllNodesWithTag("ProfileImage").fetchSemanticsNodes().isNotEmpty()
-    }
+    // Verify the specific parking is removed from favorites
+    composeTestRule.onNodeWithText("Rue de la paix").assertDoesNotExist()
+  }
 
-    // Verify profile image is clickable in edit mode
-    composeTestRule.onNodeWithTag("ProfileImage").assertHasClickAction()
+  @Test
+  fun testCancelRemoveFavoriteParking() {
+    composeTestRule.waitForIdle()
+
+    // Click the star icon to show the confirmation dialog
+    composeTestRule.onNodeWithTag("FavoriteToggle_0").performClick()
+    composeTestRule.waitForIdle()
+
+    // Cancel removal
+    composeTestRule.onNodeWithText("Cancel").performClick()
+    composeTestRule.waitForIdle()
+
+    // Verify the specific parking is still in favorites
+    composeTestRule.onNodeWithText("Rue de la paix").assertIsDisplayed()
   }
 
   @Test
-  fun testRemoveFavoriteParking() {
-    // Wait for favorite parkings to be displayed
-    composeTestRule.waitUntil(timeoutMillis = 5000) {
-      composeTestRule.onAllNodesWithTag("FavoriteToggle_0").fetchSemanticsNodes().isNotEmpty()
-    }
+  fun testRemoveAllFavoriteParkings() {
```

Comment:
For this test, you're always removing the first parking (because upon removing, the index get modify). Could you also test the case where you remove a parking in the middle as well as the last parking, and check for the indexes ?

Reply:
I added a test where i remove the middle parking, assert the 1st and 3rd (now 2nd) are still here. And then I remove the 3rd (now 2nd) and assert the 1st is still displayed. Is that what you wanted ?